api/v1alpha1/externaldns_types.go

```diff
@@ -405,8 +401,23 @@ type ExternalDNSSource struct {
 	//
 	// +kubebuilder:validation:Required
 	// +kubebuilder:default:=Ignore
-	// +required
+	// +optional
 	HostnameAnnotationPolicy HostnameAnnotationPolicy `json:"hostnameAnnotation"`
+
+	// FQDNTemplate sets a templated string that's used to generate DNS names
+	// from sources that don't define a hostname themselves.
+	// Multiple global FQDN templates are possible.
+	//
+	// Should not be empty when HostnameAnnotationPolicy is set to Ignore.
```

Comment:
should not or must not?

Reply:
Is this honored when HostnameAnnotationPolicy is not Ignore?

Reply:
Should not. Because it _must be_ not empty for the source of type `Service` and _can be_ empty for the source type `OpenShiftRoute`. We have a corresponding [validating webhook check](https://github.com/openshift/external-dns-operator/blob/main/api/v1alpha1/externaldns_webhook.go#L102).

When `HostnameAnnotationPolicy` is not `Ignore` this field can be empty regardless of the source type.

Reply:
> Should not. Because it must be not empty for the source of type Service and can be empty for the source type OpenShiftRoute. We have a corresponding validating webhook check.

You should be crisp about the cases.